### PR TITLE
Assign license via BIQ-IQ

### PR DESCRIPTION
**Pull request by zhaoqin-github**

### CODE REVIEW

File: app/waf/src/models/adc.model.ts [hunk 1]
```diff
@@ -217,16 +214,18 @@ export class Adc extends CommonEntity {
   })
   lastErr: ConfigTypes['lastErr'];
 
+  username: string = 'admin';
```
Comment (jikui) [inline app/waf/src/models/adc.model.ts]: The username and password will be default?

File: app/waf/src/controllers/adc.controller.ts [hunk 2]
```diff
@@ -153,6 +147,11 @@ export class AdcController extends BaseController {
           await runWithTimer(`${adc.id}.onboard`, () =>
             this.onboard(adc, addonReq),
           );
+
```
Comment (jikui) [inline app/waf/src/controllers/adc.controller.ts]: The logic is for the Hardware bigip?

File: app/waf/src/controllers/adc.controller.ts [hunk 3]
```diff
@@ -116,7 +110,7 @@ export class AdcController extends BaseController {
       tenantId: await this.reqCxt.get(WafBindingKeys.Request.KeyTenantId),
       reqId: this.reqCxt.name,
     };
-    this.adcStCtr = new AdcStateCtrlr(adc, addonReq);
+    this.adcStCtr = new AdcStateCtrlr(adc, addonReq, this.asgMgr);
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: It's needless to pass the 3rd parameter to `AdcStateCtrlr`. 
If you are concerning the unit tests and want to stub it up with Sinon, why not stub it with:
`let stubInst = sinon.stub(AdcStateCtrlr.prototype, "some-function"); .. stubInst.callfake(..); ..`.
Have you searched online for it, or it doesn't work for you?

File: app/waf/src/controllers/adc.controller.ts [hunk 4]
```diff
@@ -153,6 +147,11 @@ export class AdcController extends BaseController {
           await runWithTimer(`${adc.id}.onboard`, () =>
             this.onboard(adc, addonReq),
           );
+
+        if (await this.adcStCtr.readyTo(AdcState.TRUSTED))
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: Why adding `setup` to the poweron/onboarding process?

File: app/waf/src/models/adc.model.ts (hunk 1 above)
Comment (zhaoqin-github) [reply app/waf/src/models/adc.model.ts]: They are default value. They can be replaced by random value, if we specify ```VE_RANDOM_PASS=true```

File: app/waf/src/controllers/adc.controller.ts (hunk 2 above)
Comment (zhaoqin-github) [reply app/waf/src/controllers/adc.controller.ts]: No logic for HW, because we postpone HW support.

**Pull request merged**